error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -615,6 +615,58 @@ Flux<S> after(Flux<T> flux) {
     }
   }
 
+  /** Prefer {@link Mono#flatMap(Function)} over more contrived alternatives. */
+  static final class MonoFlatMapIdentity<S, T> {
+    @BeforeTemplate
+    Mono<T> before(Mono<S> mono, Function<S, Mono<T>> function) {
```

Comment:
This matches more, and is consistent with the relevant parameter types:
```suggestion
    Mono<T> before(Mono<S> mono, Function<? super S, ? extends Mono<? extends T>> function) {
```